ocaml/xapi/quicktest_import_raw_vdi.ml

```diff
@@ -0,0 +1,55 @@
+(*
+ * Copyright (C) 2017 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+
+open Client
+open Quicktest_common
+open Http
+
+let http request f =
+  let open Xmlrpc_client in
+  let transport =
+    if !using_unix_domain_socket
+    then Unix Xapi_globs.unix_domain_socket
+    else SSL(SSL.make ~use_fork_exec_helper:false (), !host, 443) in
+  with_transport transport (with_http request f)
+
+let import_raw_vdi ~session_id ~task_id f =
+  let req = Http.Request.make ~version:"1.0"
+      ~user_agent:"quicktest"
+      ~query:[
+        "session_id", Ref.string_of session_id;
+        "sr_id","fake_sr";
+        "task_id", Ref.string_of task_id
+      ]
+      Http.Put "/import_raw_vdi" in
+  http req (fun (_, fd) -> f fd)
```

Comment:
Oh yes, that's right, my comment is wrong: this function that gets called raises an exception because we return the `400` HTTP status code. So sadly we cannot check the exact error code of the response with the current implementation of `http_rpc_recv_response` in `xen-api-libs-transitional`. Maybe in the future we should raise a `Http_error` for the HTTP 400 error code as well from that function.